tools/twix/src/panels/plot.rs

```diff
@@ -91,14 +101,75 @@ impl LineData {
                                 let value = lua_function
                                     .call::<_, f64>(self.lua.to_value(value))
                                     .unwrap_or(f64::NAN);
-                                [i as f64, value]
+                                [(buffer_size + i - buffered_values.len()) as f64, value]
```

Comment:
In my mind it makes more sense to write
```suggestion
                                [(buffer_size - buffered_values.len() + i) as f64, value]
```

Mayhaps even create a variable to store the padding of `buffer_size - buffer_values.len()`.

Reply:
I implemented Maxis solution from below instead